### PR TITLE
Enable setting PSSession Name when using SSHTransport and added Transport property

### LINKED ISSUE #2426: Runspace names listed by Get-Runspace should have meaningful names when possible
Currently when a new runspace is created its default name is "Runspace#".  The only exception to this is the runspace created on a host process for IPC and it is named "RemoteHost".

We should provide runspace names with meaning when possible.  This would make it easier for users to determine which runspace in the list is the one they want to examine or debug.

For example:
1. The first runspace created in a session with a PSHost should be named "HostRunspace" (now it is "Runspace1").
2. The IPC host runspace should be renamed to "IPCHostRS".
3. The runspace proxy created for a WinRM remote session should be "WinRMRunspace#".
4. The runspace proxy created for an SSH remote session should be "SSHRunspace#".
5. Internal runspaces created by the PowerShell engine could be something like "InternalRS#", or "PSTempRS#".
6. Workflow runspaces could be "PSWorkflowRS#".

### LINKED ISSUE #5951: Bug: PSSession Name is not properly set when using "SSHHost" parameter set name
### Reproduction Steps

```
PS /Users/tsulli> New-PSSession -SSHTransport -Name asdf -Port 22 -HostName x.x.x.x -UserName AdminUser -KeyFilePath /Users/tsulli/Spock.pem
```

### Expected Result

The PSSession object has the name specified in the `-Name` parameter of the `New-PSSession` command.

### Actual Result

The PSSession object does not have the name specified in the `-Name` parameter of the `New-PSSession` command.

```
 Id Name            ComputerName    ComputerType    State         ConfigurationName     Availability
 -- ----            ------------    ------------    -----         -----------------     ------------
  9 SSH9            x.x.x.x  RemoteMachine   Opened        DefaultShell             Available
```

Cheers,
Trevor Sullivan

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

When using `New-PSSession -SSHTransport -Name ...` the `Name` parameter wasn't being used when creating the remoterunspace so an automatically generated one was always created.
Fix is to explicitly pass the transport name when generating a runspace name including other transports like VMBus and Container.

Refactor runspace name generation code to produce "RunspaceN" and added a Transport property.

Test will have to wait until we have infrastructure for remoting tests and `Enable-SSHRemoting` work is complete.

Fix https://github.com/PowerShell/PowerShell/issues/5951
Fix https://github.com/PowerShell/PowerShell/issues/2426

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [X] Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed - Issue link:
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs [hunk 1]
```diff
@@ -1081,7 +1082,14 @@ private List<RemoteRunspace> CreateRunspacesForSSHHostParameterSet()
                     this.KeyFilePath,
                     this.Port);
                 var typeTable = TypeTable.LoadDefaultTypeFiles();
-                remoteRunspaces.Add(RunspaceFactory.CreateRunspace(sshConnectionInfo, this.Host, typeTable) as RemoteRunspace);
+                int rsId;
+                string rsName = GetRunspaceName(index, out rsId);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs]: We could remove line 1085:
```c#
string rsName = GetRunspaceName(index, out int rsIdUnused); 
```

File: src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs [hunk 2]
```diff
@@ -1099,7 +1108,14 @@ private List<RemoteRunspace> CreateRunspacesForSSHHostHashParameterSet()
                     sshConnection.KeyFilePath,
                     sshConnection.Port);
                 var typeTable = TypeTable.LoadDefaultTypeFiles();
-                remoteRunspaces.Add(RunspaceFactory.CreateRunspace(sshConnectionInfo, this.Host, typeTable) as RemoteRunspace);
+                int rsId;
+                string rsName = GetRunspaceName(index, out rsId);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs]: The same.

**Review by iSazonov — COMMENTED** [review]
Leave a comment

File: src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs (hunk 1 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs]: Yeah, can't believe I missed that.

File: src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs (hunk 2 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs]: will fix

**Comment by SteveL-MSFT** [comment]
Need to redo this fix.  `GetRunspaceName` assumes WinRM.

**Comment by iSazonov** [comment]
@SteveL-MSFT I found related Issue #2426 - can we address it in the PR (in part of the names at least)?

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs [hunk 3]
```diff
@@ -679,7 +679,7 @@ private List<RemoteRunspace> CreateRunspacesWhenRunspaceParameterSpecified()
 
                         // Create new remote runspace with name and Id.
                         int rsId;
-                        string rsName = GetRunspaceName(rsIndex, out rsId);
+                        string rsName = GetRunspaceName("WSMan", rsIndex, out rsId);
```
[the remark this answers is not in the document]
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs]: Makes sense.  Will fix.

**Comment by SteveL-MSFT** [comment]
@iSazonov based on #2426, it seems that the autogenerated name can actually be simplified to just `RunspaceN` and instead of putting the transport into the name, have it as a separate property.  I can do that as part of this PR.

File: src/System.Management.Automation/engine/remoting/client/remoterunspaceinfo.cs [hunk 4]
```diff
@@ -192,6 +192,16 @@ public Runspace Runspace
             }
         }
 
+        /// <summary>
+        /// Name of the transport used.
+        /// </summary>
+        public String Transport
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/client/remoterunspaceinfo.cs]: Minor comment: You could use the new property accessor expression syntax here
```
public String Transport => GetTransportName();
```

**Review by PaulHigin — APPROVED** [review]

**Pull request merged**